pkg/asset/machines/aws/worker.go

```diff
@@ -23,60 +30,70 @@ type MachineConfig struct {
 	Machine     types.AWSMachinePoolPlatform
 }
 
-// WorkerMachineSetTmpl is template for worker machineset.
-var WorkerMachineSetTmpl = template.Must(template.New("aws-worker-machineset").Parse(`
-apiVersion: cluster.k8s.io/v1alpha1
-kind: MachineSet
+// WorkerMachineSetsTmpl is template for worker machinesets.
+var WorkerMachineSetsTmpl = template.Must(template.New("aws-worker-machinesets").Parse(`
+{{- $c := . -}}
+kind: List
+apiVersion: v1
 metadata:
-  name: {{.ClusterName}}-worker-0
-  namespace: openshift-cluster-api
-  labels:
-    sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
-    sigs.k8s.io/cluster-api-machine-role: worker
-    sigs.k8s.io/cluster-api-machine-type: worker
-spec:
-  replicas: {{.Replicas}}
-  selector:
-    matchLabels:
-      sigs.k8s.io/cluster-api-machineset: worker
-      sigs.k8s.io/cluster-api-cluster: {{.ClusterName}}
-  template:
-    metadata:
-      labels:
+  resourceVersion: ""
+  selfLink: ""
+items:
+{{- range $index,$instance := $c.Instances}}
+- apiVersion: cluster.k8s.io/v1alpha1
+  kind: MachineSet
+  metadata:
+    name: {{$c.ClusterName}}-worker-{{$index}}
```

Comment:
Is it possible to use the pool name instead of hardcoding worker here?

Reply:
we are currently generating the ign for only workers. So we can only support works for now.